### ISSUE TITLE
Software renderer, when at 2x, reverts to 1x for some scenes in legend of dragoon

### ISSUE THREAD

**Issue by pharmville** [libretro/beetle-psx-libretro#201]
I have a decent pc so I always use the software renderer with 2x internal resolution. Now that I'm playing Legend of Dragoon, I notice there are a few events in the game that causes the emulator to switch to 1x internal resolution. Once the triggering game event finishes, it returns to 2x resolution. The easiest way to see this effect: Go to battle and transform into a dragoon. During the quick dragoon transformation, everything will be rendered in 1x and look super blocky, but after the transformation is complete, it is 2x again.

**Comment by hizzlekizzle** [comment]
I suspect that's a framebuffer effect, which is always rendered at 1x. That is, nothing to be done about it, it's just how things work.

**Comment by simias** [comment]
Yeah, you have this type of issues in a bunch of games. What happens is that the game loads the framebuffer in main RAM to do some effects with the CPU then re-uploads the modified texture to the GPU for display.

There's no easy way to work around that, obviously when the game reads the framebuffer in RAM it allocates a certain buffer size, we can't just send a bigger image and expect things to work. So any time the framebuffer is loaded into main RAM we dowscale it back to 1x to make the game happy.

There's really no simple way to work around that I'm afraid. Maybe we could look at what N64 emulators are doing since on this console the framebuffers are always directly accessible from the CPU and therefore these effects are probably more common.
